### PR TITLE
Makefile: fix seccomp notify compilation issues

### PR DESCRIPTION
**Pull request by haircommander**
on older kernels we need to also check if the proper headers are installed, in addition to the seccomp version

also, unify the syntax of conditional compilation variables

### CODE REVIEW

**Comment by haircommander** [comment]
fixes https://github.com/containers/conmon/issues/275

**Comment by rhatdan** [comment]
LGTM

**Review by saschagrunert — APPROVED** [review]
LGTM

File: Makefile [hunk 1]
```diff
@@ -38,15 +38,15 @@ override CFLAGS += $(shell $(PKG_CONFIG) --cflags glib-2.0) -DVERSION=\"$(VERSIO
...
-else ifeq ($(shell $(PKG_CONFIG) --exists libsystemd && echo "0" || echo "1"), 0)
+	override CFLAGS += $(shell $(PKG_CONFIG) --cflags libsystemd-journal) -D USE_JOURNALD=1
+else ifeq ($(shell $(PKG_CONFIG) --exists libsystemd && echo "0"), 0)
 	override LIBS += $(shell $(PKG_CONFIG) --libs libsystemd)
-	override CFLAGS += $(shell $(PKG_CONFIG) --cflags libsystemd) -D USE_JOURNALD=0
+	override CFLAGS += $(shell $(PKG_CONFIG) --cflags libsystemd) -D USE_JOURNALD=1
 endif
 
-ifeq ($(shell $(PKG_CONFIG) --atleast-version 2.5.0 libseccomp && echo "0" || echo "1"), 0)
+ifeq ($(shell $(PKG_CONFIG) --atleast-version 2.5.0 libseccomp && grep -rq "seccomp_notif_sizes" /usr/include && echo "0"), 0)
```
Comment (giuseppe) [inline Makefile:49]: this could be expensive, and detect the string in other unrelated header files.

What do you think about replacing it with `printf "#include <linux/seccomp.h>\nvoid main(){struct seccomp_notif_sizes s;}" | cc -x c - -o /dev/null` ?

Comment (rhatdan) [reply Makefile:49]: SGTM

Comment (haircommander) [reply Makefile]: updated as suggested!
